MediaBrowser.MediaEncoding/Subtitles/JsonWriter.cs

```diff
@@ -1,27 +1,39 @@
 using System.IO;
-using System.Text;
+using System.Text.Json;
 using System.Threading;
 using MediaBrowser.Model.MediaInfo;
-using MediaBrowser.Model.Serialization;
 
 namespace MediaBrowser.MediaEncoding.Subtitles
 {
+    /// <summary>
+    /// JSON subtitle writer.
+    /// </summary>
     public class JsonWriter : ISubtitleWriter
     {
-        private readonly IJsonSerializer _json;
-
-        public JsonWriter(IJsonSerializer json)
-        {
-            _json = json;
-        }
-
+        /// <inheritdoc />
         public void Write(SubtitleTrackInfo info, Stream stream, CancellationToken cancellationToken)
         {
-            using (var writer = new StreamWriter(stream, Encoding.UTF8, 1024, true))
+            using (var writer = new Utf8JsonWriter(stream))
             {
-                var json = _json.SerializeToString(info);
+                var trackevents = info.TrackEvents;
+                writer.WriteStartArray("TrackEvents");
+
+                for (int i = 0; i < trackevents.Count; i++)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    var current = trackevents[i];
+                    writer.WriteStartObject();
+
+                    writer.WriteString("Id", current.Id);
+                    writer.WriteString("Text", current.Text);
+                    writer.WriteNumber("StartPositionTicks", current.StartPositionTicks);
+                    writer.WriteNumber("EndPositionTicks", current.EndPositionTicks);
+
+                    writer.WriteEndObject();
+                }
```

Comment:
why hand-roll instead of using `json.SerializeToString()`?

Reply:
This can get cancelled with the cancellation token, just like all the other subtitle writers.

Reply:
I thought so, but wanted to be sure.

Reply:
But cannot we use json to encode a `trackevents[i]` element as a whole instead of packing it piece by piece?

Reply:
We could... but this is a cleaner solution imo, although it’s a bit longer.